src/code/cutscene_camera.c

```diff
@@ -0,0 +1,379 @@
+#include "global.h"
+
+extern CutsceneCamera* sCurCsCamera;
+
+// function declarations
+s16 func_80161180(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_8016237C(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_8016253C(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_80162A50(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_801623E4(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_80161C20(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_80161E4C(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_801620CC(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+s16 func_80163334(Vec3f* pos, f32* fov, s16* roll, CsCmdCamPoint* point, CsCmdCamMisc* misc,
+                  CutsceneCameraInterp* interp);
+f32 func_80163660(Actor* actor);
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80161180.s")
+
+/**
+ * Initializes Cutscene Camera Info
+ */
+s32 CutsceneCamera_Init(Camera* camera, CutsceneCamera* csCamera) {
+    csCamera->camera = camera;
+
+    csCamera->nextSplineTimer = csCamera->updateSplineTimer = 0;
+    csCamera->cmdIndex = 0;
+    csCamera->splineIndex = 0xFFFF;
+    csCamera->splineNeedsInit = true;
+    csCamera->state = CS_CAM_STATE_UPDATE_ALL;
+
+    sCurCsCamera = csCamera;
+
+    __osMemset(&csCamera->eyeInterp, 0, sizeof(CutsceneCameraInterp));
+    __osMemset(&csCamera->atInterp, 0, sizeof(CutsceneCameraInterp));
+
+    csCamera->eyeInterp.unk_2D = csCamera->atInterp.unk_2D = 7;
+
+    return 1;
+}
+
+s16 (*CutsceneCamera_Interpolate(u8 interpType))(Vec3f*, f32*, s16*, CsCmdCamPoint*, CsCmdCamMisc*,
```

Comment:
monkaS
Have you considered making a typedef for this callback?

Reply:
These typedefs always confuse me, could you help a fellow out? What would the typedef look like?

Reply:
If I'm not wrong it should look like this (assuming you want to name it `InterpolateCallback`)
```c
typedef s16 (*InterpolateCallback)(Vec3f*, f32*, s16*, CsCmdCamPoint*, CsCmdCamMisc*, CutsceneCameraInterp*);
```

Reply:
Nice, thanks!